src/PlusDataCollection/PointGrey/vtkPlusSpinnakerVideoSource.cxx

```diff
@@ -0,0 +1,807 @@
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::SetSharpeningMode(SHARPENING_MODE sharpMode)
+{
+  this->SharpeningMode = sharpMode;
+  return PLUS_SUCCESS;
+}
+
+//-----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::SetSharpeningAmount(float sharpeningAmount)
+{
+  if (sharpeningAmount < 0 || sharpeningAmount > 0.25)
+  {
+    LOG_WARNING("Requested invalid sharpening value of " << sharpeningAmount << ", sharpening has been turned off. Valid sharpening values are in the range [0, 0.25].");
+    this->SharpeningMode = SHARPENING_OFF;
+    return PLUS_FAIL;
+  }
+  this->SharpeningAmount = sharpeningAmount;
+  return PLUS_SUCCESS;
+}
+
+//-----------------------------------------------------------------------------
+// verify that if any manual settings are enabled then their corresponding values have also been set
+PlusStatus vtkPlusSpinnakerVideoSource::CheckCameraParameterValidity()
+{
+  PlusStatus retVal = PLUS_SUCCESS;
+
+  // exposure
+  if (this->ExposureMode == EXPOSURE_TIMED && this->ExposureMicroSec == FLAG_EXPOSURE_MICROSEC)
+  {
+    LOG_ERROR("Failed to configure SpinnakerVideoSource: ExposureMode is set to \"TIMED\", but no ExposureMicroSec attribute is provided.");
+    retVal = PLUS_FAIL;
+  }
+
+  // gain
+  if (this->GainMode == GAIN_MANUAL && this->GainDB == FLAG_GAIN_DB)
+  {
+    LOG_ERROR("Failed to configure SpinnakerVideoSource: GainMode is set to \"MANUAL\", but no GainDB attribute is provided.");
+    retVal = PLUS_FAIL;
+  }
+
+  // white balance
+  if (this->WhiteBalanceMode == WB_MANUAL && this->WhiteBalanceRed == FLAG_WHITE_BALANCE)
+  {
+    LOG_ERROR("Failed to configure SpinnakerVideoSource: WhiteBalanceMode is set to \"MANUAL\", but no WhiteBalanceRed attribute is provided.");
+    retVal = PLUS_FAIL;
+  }
+  if (this->WhiteBalanceMode == WB_MANUAL && this->WhiteBalanceBlue == FLAG_WHITE_BALANCE)
+  {
+    LOG_ERROR("Failed to configure SpinnakerVideoSource: WhiteBalanceMode is set to \"MANUAL\", but no WhiteBalanceBlue attribute is provided.");
+    retVal = PLUS_FAIL;
+  }
+
+  // sharpening
+  if (this->SharpeningMode == SHARPENING_MANUAL && this->SharpeningAmount == FLAG_SHARPENING)
+  {
+    LOG_ERROR("Failed to configure SpinnakerVideoSource: SharpeningMode is set to \"MANUAL\", but no SharpeningAmount attribute is provided.");
+    retVal = PLUS_FAIL;
+  }
+
+  return retVal;
+}
+
+//-----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::ReadConfiguration(vtkXMLDataElement* rootConfigElement)
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::ReadConfiguration(vtkXMLDataElement* rootConfigElement)");
+
+  XML_FIND_DEVICE_ELEMENT_REQUIRED_FOR_READING(deviceConfig, rootConfigElement);
+
+  // camera params
+  XML_READ_SCALAR_ATTRIBUTE_REQUIRED(int, CameraNumber, deviceConfig);
```

Comment:
Would you consider making camera number optional and using the first available camera if not defined?